Zilon.Core/Zilon.Core.Tests/World/BiomInitializerTests.cs

```diff
@@ -0,0 +1,185 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using FluentAssertions;
+
+using Moq;
+
+using NUnit.Framework;
+
+using Zilon.Core.MapGenerators;
+using Zilon.Core.Schemes;
+using Zilon.Core.Tactics;
+using Zilon.Core.Tests.Common.Schemes;
+
+namespace Zilon.Core.World.Tests
+{
+    [TestFixture()]
+    [Category("integration")]
+    public class BiomInitializerTests
+    {
+        [Test()]
+        public async Task BuildLinearBiomGraph()
+        {
+            // ARRANGE
+
+            var sectorGeneratorMock = new Mock<ISectorGenerator>();
+            sectorGeneratorMock.Setup(x => x.GenerateAsync(It.IsAny<ISectorNode>()))
+                .Returns<ISectorNode>(scheme =>
+                {
+                    return Task.Run(() =>
+                    {
+                        var sectorMock = new Mock<ISector>();
+                        var sector = sectorMock.Object;
+                        return sector;
+                    });
+                });
+
+            var sectorGenerator = sectorGeneratorMock.Object;
+            var locationSchemes = CreateBiomSchemes();
+
+            var schemeRoundCounter = 0;
+            var rollerMock = new Mock<IBiomeSchemeRoller>();
+            rollerMock.Setup(x => x.Roll()).Returns(() =>
+            {
+                schemeRoundCounter++;
+                if (schemeRoundCounter >= locationSchemes.Length)
+                {
+                    schemeRoundCounter = 0;
+                }
+
+                var scheme = locationSchemes[schemeRoundCounter];
+
+                return scheme;
+            });
+            var roller = rollerMock.Object;
+
+            var schemeServiceMock = new Mock<ISchemeService>();
+            schemeServiceMock.Setup(x => x.GetSchemes<ILocationScheme>())
+                .Returns(locationSchemes);
+            schemeServiceMock.Setup(x => x.GetScheme<ILocationScheme>(It.IsAny<string>()))
+                .Returns<string>(sid => locationSchemes.Single(scheme => scheme.Sid == sid));
+            var schemeService = schemeServiceMock.Object;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'schemeService'.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5260137)